src/ipld/cbor.js

```diff
@@ -0,0 +1,17 @@
+import * as API from '../api.js'
+import * as CBOR from '@ipld/dag-cbor'
+
+/**
+ * @type {API.MulticodecCode<typeof CBOR.code, typeof CBOR.name>}
+ */
+export const code = CBOR.code
```

Comment:
`API` is only used from types. Move to jsdoc `@typedef import`?

```suggestion
import * as CBOR from '@ipld/dag-cbor'

/**
 * @typedef {import('../api.js').MulticodecCode} MulticodecCode
 * @type {MulticodecCode<typeof CBOR.code, typeof CBOR.name>}
 */
export const code = CBOR.code
```

Reply:
Unfortunately it's not the same, as with suggested changes `MulticodecCode` is no longer generic type. If you want to make it generic then you have to also define `@template` tags and etc.. which quickly gets out of hand. For more context see https://github.com/microsoft/TypeScript/issues/41825

From my experience empty `api.js` module along with `api.ts` is a most pragmatic solution to this problem.